.harness/reconcile_dependencies_for_all_envs_based_on_parent_folder.yaml

```diff
@@ -0,0 +1,163 @@
+
+                      # Navigate to environment directory
+                      ENV_DIR=<+pipeline.variables.env_folder>
+                      TARGET_DIR=${ENV_DIR}
+
+                      # Capture the filtered directories into a variable
+                      CHANGED_DIRS=$(git diff --name-only "$current_branch" "$target_branch" \
+                        | grep '/' \
+                        | xargs -n1 dirname \
+                        | grep "^$TARGET_DIR" \
+                        | while read -r DIR; do
+                            REST=${DIR#"$TARGET_DIR/"}
+                            if [ -z "$REST" ]; then
+                              echo "$DIR"
+                            else
+                              FIRST_LEVEL=$(echo "$REST" | cut -d'/' -f1)
+                              echo "$TARGET_DIR/$FIRST_LEVEL"
+                            fi
+                          done \
+                      | sort -u)
+
+                      # Output the list stored in the variable
+                      echo "=== Changed environments ==="
+                      echo "$CHANGED_DIRS"
+                      echo "=== === === === === === ==="
+
+
+                      # Iterate over each directory in the list
+                      for DIR in $CHANGED_DIRS; do
+                        # Run the script for each directory
+                        cd ${DIR} || exit 1
+                        echo "Running: bash ${ROOT_DIR}/tools/reconcile_dependencies.sh $DIR"
+                        bash ${ROOT_DIR}/tools/reconcile_dependencies.sh "$DIR"
+
+                        # This could be a searate step
+                        echo "Checking whether Environment Version ID should be updated"
+                        git show master:./env_info.json | grep -n 'environmentVersionId' > /tmp/old_env_version_id.txt
+                        git show HEAD:./env_info.json | grep -n 'environmentVersionId' > /tmp/new_env_version_id.txt
+                        # If env version was not updated, update
+                        if diff /tmp/old_env_version_id.txt /tmp/new_env_version_id.txt > /dev/null; then
+                          echo "Updating Environment Version ID"
+                          python3 ${ROOT_DIR}/tools/env_version_update.py --file ./env_info.json
+                        fi
+
+
+                        cd -
+                      done
+
+                      # Run dependency reconciliation if there are file changes
+                      #if [ -n "$(echo $(git diff --name-only master -- ./requirements.*))" ]; then
+                      #  echo "Requirements changes detected in $ENV_DIR. Running reconciliation..."
+                      #  bash ${ROOT_DIR}/tools/reconcile_dependencies.sh
+                      #else
+                      #  echo "No requirements changes detected. Skipping reconciliation."
+                      #  exit 0  # Exit early if no changes
+                      #fi
```

Comment:
As I understand the changes are checked only for version id. The requirements (.in?) files are always auto generated?

Reply:
I removed this commented out code.
Now we always run pip-compile if there are any changes in the environment. 
E.g. you publish new drum, then do any change(for example: change env version id or do some change in requirements.txt) to trigger reconciliation.